olive/passes/onnx/transformer_optimization.py

```diff
@@ -217,10 +217,12 @@ def _run_for_config(
                 "OrtTransformersOptimization.config"
             )
 
-        output_model_path = ONNXModel.resolve_path(os.path.join(output_model_path, os.path.basename(model.model_path)))
+        output_model_path = Path(output_model_path)
+        if output_model_path.suffix != ".onnx":
+            output_model_path = output_model_path / Path(model.model_path).stem
+        output_model_path = ONNXModel.resolve_path(output_model_path)
```

Comment:
Double check: if the ranked model is model_xx.onnx, then in this change, the output path is `output_model_paht/model_xx/model.onnx`. Is that the one you expect?

Reply:
The paths are correct. The input is not a folder but rather a file path like `output_model_path/model_xx.onnx`. Here's the output from local run -

```
│   ├── [       4096]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169
│   │   └── [       4096]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model
│   │       ├── [    3250819]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_00.onnx
│   │       ├── [ 3762823168]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_00.onnx.data
│   │       ├── [    3250819]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_01.onnx
│   │       ├── [ 3762823168]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_01.onnx.data
│   │       ├── [    3250819]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_02.onnx
│   │       ├── [ 3762823168]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_02.onnx.data
│   │       ├── [    3250819]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_03.onnx
│   │       └── [ 3762823168]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169/output_model/model_03.onnx.data
│   ├── [       3121]  cache/models/2_OrtTransformersOptimization-1-d77132b8b5bfe76b03d9da8112411169.json
```

Reply:
Okay, I see, for distributed case, the `output_model_path` in line220 is a file ends with `.onnx`.

But for the general case, the input is a folder, for example the bert case, then the `output_model_path` in Line223 will be `output_model_path/model/model.onnx`.
![image](https://github.com/microsoft/Olive/assets/13343117/44800a5e-5cda-448c-a46d-a244d21d104a)

I think that line22 should be:
`output_model_path = output_model_path / Path(model.model_path).name`

Reply:
`output_model_path = ONNXModel.resolve_path(output_model_path)` will always append `model.onnx` if the input does not ends with `.onnx`.
![image](https://github.com/microsoft/Olive/assets/13343117/5c37cf6b-ff58-4e9c-b858-106956267b51)

Reply:
You are correct. I was trying to maintain the old behavior. Equivalent of `os.path.basename` is `pathlib.Path.name`.